internal/controller/glideinset_controller.go

```diff
@@ -125,6 +132,49 @@ func finalizeGlideinSet(glideinSet *gmosv1alpha1.GlideinSet) {
 	removeCollectorClient(glideinSet)
 }
 
+// Update the GlideinSet's RemoteManifest field based on new data in its Glidein Manager's git repository
+func (pr *PilotSetReconcileState) applyGitUpdate(gitUpdate gmosClient.RepoUpdate) error {
+	log := log.FromContext(pr.ctx)
+	log.Info("Got repo update!")
+
+	log.Info("Updating GlideinSet remote manifest")
+	err := applyUpdateToResource(pr, RNBase, &gmosv1alpha1.GlideinSet{}, &GlideinSetGitUpdater{gitUpdate: &gitUpdate})
+	if !updateErrOk(err) {
+		return err
+	}
+	return nil
+}
+
+// Update the GlideinManagerPilotSet's children based on new data in its Glidein Manager's
+// secret store
+func (pu *PilotSetReconcileState) applySecretUpdate(secSource gmosv1alpha1.PilotSetSecretSource, sv gmosClient.SecretValue) error {
+	log := log.FromContext(pu.ctx)
+	log.Info("Secret updated to version " + sv.Version)
+	return applyUpdateToResource(pu, RNTokens, &corev1.Secret{}, &TokenSecretValueUpdater{secSource: &secSource, secValue: &sv})
+}
+
+// Retrieve the current
+func (pr *PilotSetReconcileState) getGitSyncState() (*gmosv1alpha1.PilotSetNamespaceConfig, error) {
+	log := log.FromContext(pr.ctx)
+	log.Info("Retrieving current Git sync state from GlideinSet")
+	currentConfig := gmosv1alpha1.GlideinSet{}
+	if err := getResourceValue(pr, RNBase, &currentConfig); err != nil {
+		log.Error(err, "Unable to retrieve Git sync state from GlideinSet")
+		return &gmosv1alpha1.PilotSetNamespaceConfig{}, err
```

Comment:
The pointer to an empty struct return might not be the best practice. Here because it points to an allocated struct, a pointer to an empty struct is not nil, which may let the it evade the nil value checking in places consuming this function.
Consider return nil instead of empty struct